src/status_im2/contexts/wallet/home/view.cljs

```diff
@@ -27,56 +27,78 @@
    {:id :collectibles :label (i18n/label :t/collectibles) :accessibility-label :collectibles-tab}
    {:id :activity :label (i18n/label :t/activity) :accessibility-label :activity-tab}])
 
+(defn get-balance
+  [balances address]
+  (.toFixed (->> balances
+                 (filter #(= (:address %) address))
+                 first
+                 :balance)
+            2))
+
+(defn account-cards
+  [accounts loading? balances]
+  (let [refactored-accounts (mapv (fn [account]
+                                    (assoc account
+                                           :type                :empty
+                                           :customization-color :blue
+                                           :on-press            #(rf/dispatch [:navigate-to
+                                                                               :wallet-accounts])
+                                           :loading?            loading?
+                                           :balance             (str "$"
+                                                                     (get-balance balances
+                                                                                  (:address account)))))
+                                  accounts)]
+    (conj refactored-accounts add-account-placeholder)))
+
+(defn- view-internal
+  [accounts]
+  (let [top          (safe-area/get-top)
+        selected-tab (reagent/atom (:id (first tabs-data)))
+        loading?     (rf/sub [:wallet-2/tokens-loading?])
+        balances     (rf/sub [:wallet-2/balances])]
+    [rn/view
+     {:style {:margin-top top
+              :flex       1}}
+     [common.top-nav/view]
+     [rn/view {:style style/overview-container}
+      [quo/wallet-overview temp/wallet-overview-state]]
+     [rn/pressable
+      {:on-long-press #(rf/dispatch [:show-bottom-sheet {:content temp/wallet-temporary-navigation}])}
+      [quo/wallet-graph {:time-frame :empty}]]
+     [rn/flat-list
+      {:style      style/accounts-list
+       :data       (account-cards accounts loading? balances)
+       :horizontal true
+       :separator  [rn/view {:style {:width 12}}]
+       :render-fn  quo/account-card}]
+     [quo/tabs
+      {:style          style/tabs
+       :size           32
+       :default-active @selected-tab
+       :data           tabs-data
+       :on-change      #(reset! selected-tab %)}]
+     (case @selected-tab
+       :assets       [rn/flat-list
+                      {:render-fn               quo/token-value
+                       :data                    temp/tokens
+                       :key                     :assets-list
+                       :content-container-style {:padding-horizontal 8}}]
+       :collectibles [collectibles/view]
+       [activity/view])]))
+
 (defn view
   []
-  (let [top          (safe-area/get-top)
-        selected-tab (reagent/atom (:id (first tabs-data)))]
+  (let [accounts (rf/sub [:profile/wallet-accounts])]
```

Comment:
@ilmotta Where do you suggest to put it?

Reply:
I added it to the `view-internal` as well. Please check if it's ok now. Just keep in mind that the `(rf/dispatch [:wallet-2/get-wallet-token accounts])` in the `view` is needed to be done in the `componentDidMount` life cycle. That's why there are two subs for it

Reply:
I don't know exactly @mmilad75, you guys have more context :)

There's another problem with the original code I forgot to mention. The event handler `:wallet-2/get-wallet-token` has access to the app db. When I see the result of a subscription being passed to the event handler it smells :), it's unnecessary because the event handler can already read the data from the coeffect `:db`. It's also more lightweight. One might say, okay, but the subscription does useful computation that I want to reuse for the event, but the better approach is just to extract to a common function and call it in the subscription and in the event handler. It's a bit hard to explain, I hope it makes sense.

```clojure
;; not so good
(defn view
  []
  (let [accounts (rf/sub [:profile/wallet-accounts])]
    (rf/dispatch [:wallet-2/get-wallet-token accounts])
    (fn []
      [view-internal])))

;; Better, change the event handler to read from the coeffect :db.
(defn view
  []
  (rf/dispatch [:wallet-2/get-wallet-token])
  (fn []
    [view-internal]))
```

> However, this (rf/dispatch [:wallet-2/get-wallet-token accounts]) should be moved to the login phase. I'm trying to move it there and remove it here

Good idea :) You're probably aware this can increase the weight on the already heavy initialization phase since we dispatch tons of events after login. It may be a good idea to use `dispatch-later` to free the event queue for a while for the more important stuff that needs to load first.

Reply:
Very good point. I removed the `accounts` param